modules/channel-smooch/src/backend/client.ts

```diff
@@ -0,0 +1,128 @@
+import * as sdk from 'botpress/sdk'
+import Smooch from 'smooch-core'
+
+import { Config } from '../config'
+
+import { Clients } from './typings'
+
+const name = 'smooch.sendMessage'
+
+export class SmoochClient {
+  private smooch: any
+  private path: string
+
+  constructor(
+    private bp: typeof sdk,
+    private botId: string,
+    private config: Config,
+    private router: any,
+    private route: string
+  ) {}
+
+  async initialize() {
+    this.smooch = new Smooch({
+      keyId: this.config.keyId,
+      secret: this.config.secret,
+      scope: 'app'
+    })
+
+    const url = (await this.router.getPublicPath()) + this.route
+    this.path = url.replace('BOT_ID', this.botId)
+
+    await this.smooch.webhooks.create({
+      target: this.path,
+      triggers: ['message:appUser']
+    })
+  }
+
+  async delete() {
+    const { webhooks } = await this.smooch.webhooks.list()
```

Comment:
Even if the library doesn't provide typings for typescript, you can write your own definition from what you find in the documentation and what is displayed in the console, it makes it much easier to debug in the future and to make changes